runtime/testrt.c

```diff
@@ -227,6 +255,7 @@ int main() {
     testStringCmp();
     testStringEq();
     testStringHash();
+    testStringConcatAssociative();
```

Comment:
We should have a bit more testing that this.

I suggest concatenating multiple strings and then checking with memcmp that the StringData of the things that are concatenated appears at the right place in the StringData of the concatenation.